yaqd_andor/_andorsdk2_ixon.py

```diff
@@ -0,0 +1,309 @@
+            self._loop.run_in_executor(None, self._check_temp_stabilized)
+        else:
+            code, self.sensor_temp = self.sdk.GetTemperature()
+            self.logger.info(f"Sensor is not cooled.  Current temp is {self.sensor_temp} C.")
+
+    def _check_temp_stabilized(self):
+        code, self.sensor_temp = self.sdk.GetTemperature()
+        diff = float(self.sensor_temp_control) - float(self.sensor_temp)
+        while (
+            np.abs(diff) > 1.0 and self.stop_update is False
+        ):  # this is a tolerance, and is subject to change
+            self.logger.info(
+                f"Sensor is cooling.  Target: {self.sensor_temp_control} C.  Current: {self.sensor_temp:0.2f} C."
+            )
+            sleep(3)
+            code, self.sensor_temp = self.sdk.GetTemperature()
+            diff = float(self.sensor_temp) - float(self.sensor_temp_control)
+        if np.abs(diff) <= 1.0:
+            self.logger.info("Sensor temp is stabilized.")
+
+    def get_dependent_hardware(self):
+        dependents = dict()
+        if self.spec_client is not None:
+            dependents = {"monochromator": self._spec_position}
+        return dependents
+
+    def set_exposure_time(self, exposure_time):
+        # Sets the exposure time in seconds (float)
+        self.stop_update = True
+        while self._busy == True:
+            sleep(0.10)
+        code = self.sdk.SetExposureTime(float(exposure_time))
+        if code != 20002:
+            raise ValueError(str(self.errorlookup(code)))
+        else:
+            self.exposure_time = exposure_time
+            self.logger.info(f"New exposure time is {self.exposure_time} sec.")
+        self.stop_update = False
+        return code
+
+    def get_exposure_time(self):
+        return self.exposure_time
+
+    def close(self):
+        self.stop_update = True
+        while self._busy == True:
+            sleep(0.50)
+        self.sdk.SetShutter(int(0), int(2), int(100), int(100))
+        sleep(1.00)
+        self.sdk.CoolerOFF()
+        """ # This portion of code is commented out unless it is found that the CCD needs
+        # to warm up before a full close is supposed to return
+        self.sensor_temp_control = int(25.0)
+        code = self.sdk.SetTemperature(self.sensor_temp_control)
+        sleep(0.20)
+        code, self.sensor_temp = self.sdk.GetTemperature()
+        diff = float(self.sensor_temp_control) - float(self.sensor_temp)
+        while np.abs(diff) > 6.0:  # this is a tolerance, and is subject to change
+            self.logger.info(
+                f"Sensor is warming to RT.  Target: {self.sensor_temp_control} C.  Current: {self.sensor_temp:0.2f} C."
+            )
+            sleep(3)
+            code, self.sensor_temp = self.sdk.GetTemperature()
+            diff = float(self.sensor_temp_control) - float(self.sensor_temp)
+        """
+        sleep(1.00)
```

Comment:
same question as above.  I really don't see how these sleep calls accomplish anything.

Reply:
camera seems to crash unless I put some time between the procedures.  Happens a lot actually.